### ISSUE TITLE
RAML generation fails if the route uri is a string that contains referenced constant

### ISSUE THREAD

**Issue by tleveque** [wisdom-framework/wisdom#500]

**Comment by cescoffier** [comment]
Yes, it is probably something we cannot really fix. The RAML processing is based on the JAVA AST, so we don't have the value of the constants.

**Comment by tleveque** [comment]
Could we detect this case and load the class to evaluate uri value ? This second compile phase can be skipped when uri value is simpler.
A second workaround would be to support only when the constants are defined into the processed class and compute uri value on the fly during Java AST parsing.

**Comment by tleveque** [comment]
I propose that my team will work on evaluating uri value with constants that are only defined into the same class as a first stepIs it ok for you ?

**Comment by cescoffier** [comment]
Cool ! 

I discuss with @barjo yesterday, and I think we would eventually look for a mechanism to merge compilation and runtime data. However, does not really help for constants, so your way is already a good step.

**Comment by barjo** [comment]
As we discuss yesterday (with Thomas ;) ) I think that's the more appropriate way to do it for now. Especially if don't want to have to handle the class during the watch mode, which can be very tricky.
